stock_rule_buy_check_stock_before/models/procurement_group.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+from odoo.osv import expression
+
+
+class ProcurementGroup(models.Model):
+    _inherit = "procurement.group"
+
+    def _is_product_available(self, product_id, quantity):
+        if quantity == 0:
+            return False
+        return product_id.free_qty >= quantity
```

Comment:
Use `immediately_usable_qty` instead of `free_qty`
Give the warehouse in context to support multi-wh
Use float_compare